src/Go/boardAnalysis/goAI.ts

```diff
@@ -63,11 +67,15 @@ export function makeAIMove(boardState: BoardState): Promise<Play> {
       }
 
       return play;
-    })
-    .finally(() => {
-      currentAITurn = null;
-      GoEvents.emit();
     });
+  }
+
+  // Once the AI moves (or the player playing as white with No AI moves),
+  // clear the isAiThinking semaphore and update the board UI.
+  Go.nextTurn.finally(() => {
+    isAiThinking = false;
+    GoEvents.emit();
```

Comment:
I still have a lingering question about the proper timing of this. Clearly, for AI moves the correct time is after the move. I assumed this would be the case for manual moves as well, which is why I encouraged reuse of this block. 

However, you still have the synchronous GoEventd.emit() in the manual case as well. I'd it covering something else?

Reply:
the "manual case" is the start of the white player's move. The board wrapper UI needs to be updated to reflect the new current player and new score. The pass button in particular changes highlighting and text if they have no legal moves, if passing will end the game, etc. Thus a call to GoEvents.emit()

The finally is the END of the white player's move. The active player has again switched, meaning the above needs to be updated once again to reflect the black player as active and update the pass button, score, and active player. 

The millisecond-exact timing doesn't matter super much since its just a UI update, not passing info to another script.